phongo_compat.h

```diff
@@ -150,23 +150,15 @@
 #elif SIZEOF_PHONGO_LONG == 4
 # define ADD_INDEX_INT64(zval, index, value) \
     if (value > INT32_MAX || value < INT32_MIN) { \
-        char *tmp; \
-        int tmp_len; \
         mongoc_log(MONGOC_LOG_LEVEL_WARNING, MONGOC_LOG_DOMAIN, "Integer overflow detected on your platform: %lld", value); \
-        tmp_len = spprintf(&tmp, 0, "%lld", value); \
-        ADD_INDEX_STRINGL(zval, index, tmp, tmp_len); \
-        efree(tmp); \
+        phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Integer overflow detected on your platform: %lld", value); \
```

Comment:
There is little value in continuing to call `mongoc_log()` if we're throwing a proper exception.